### ISSUE TITLE
Feature Request: select files with wildcard or via regex

### ISSUE THREAD

**Issue by mqnc** [egorovsa/vscode-folder-colorizer#10]
I have all these configuration files in many subproject folders:
```
.dockerignore
.flake8
.gitignore
.gitmodules
.isort.cfg
.style.yapf
```
and I'd love to be able to color them all at once in some unobtrusive color...
I think it would be okay if this can only be done manually in the settings.json, maybe like this:
```
		{
			"folderPath": "*/.flake8/",
			"color": "foldercolorizer.color_996633"
		},
```

**Comment by egorovsa** [comment]
Hello. Could you please clarify? Do you want to colorize all files, such as .flake8 etc., regardless of their location or level in the folder structure of the project? I think it's possible and will add this functionality soon!

**Comment by mqnc** [comment]
That is exactly what I mean. With wildcards or regex we can even go beyond specific filenames (like .flake8) and maybe have all files of a certain type (like *.bak for backup files of some programs) be a specific color.

However, this would mean that there could be conflicts. Like if you have the two rules:
- all files that match *.jpg should be red
- all files that match tree.* should be green
then what color should tree.jpg be?

I would say in that case the rules in settings.json are just applied in order and the later ones win.

**Comment by iamkarlson** [comment]
Hi, is there any solution for that? I would love to add some wildcard path to my global `settings.json` (like `*/src/` or `*/tests/`). I see in #14 you are waiting for some API from vscode, perhaps there's something can be done in the meantime?

**Comment by egorovsa** [comment]
Hello.. I will try to publish this changes as soon as possible, more over it's previously done. 
I need just to check this again and if all's ok, I will publish it.
So yes you will be able to colorize specific extensions or folder path patterns. ;)

**Comment by iamkarlson** [comment]
I just found out, that you don't need to specify the whole path and it actually works fine with a global `settings.json`:
```
{
    .......................................................................
    "folder-color.pathColors": [
        {
            "folderPath": "tests/",
            "color": "foldercolorizer.color_0066cc"
        },
        {
            "folderPath": "src/",
            "color": "foldercolorizer.color_336600"
        }
    ]
}
```

It actually works much better cause for some projects I have `tests` inside `src` and it still colorize them differently! Nice!

**Comment by egorovsa** [comment]
Yes, you're right. It will work since the pattern is matching; however, 
be careful "folderPath": "src/", will also take in account the folders with name "mysrc". In order to avoid this use pattern  "folderPath": "/src/", and for tests like "folderPath": "/tests/",
it's also not available in the context menu. I will add this option to the context menu as well.

**Comment by egorovsa** [comment]
Also note: in case there is more specific path, all the "global patterns" like "folderPath": "/tests/", will be overloaded. I will try to handle it.
